internal/plumbing/ticks.go

```diff
@@ -0,0 +1,158 @@
+package plumbing
+
+import (
+	"log"
+	"time"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/hercules.v9/internal/core"
+)
+
+// TicksSinceStart provides relative tick information for every commit.
+// It is a PipelineItem.
+type TicksSinceStart struct {
+	core.NoopMerger
+	remote       string
+	tickSize     time.Duration
+	tick0        *time.Time
+	previousTick int
+	commits      map[int][]plumbing.Hash
+}
+
+const (
+	// DependencyTick is the name of the dependency which DaysSinceStart provides - the number
+	// of ticks since the first commit in the analysed sequence.
+	DependencyTick = "tick"
+
+	// FactCommitsByTick contains the mapping between day indices and the corresponding commits.
+	FactCommitsByTick = "TicksSinceStart.Commits"
+
+	// ConfigTicksSinceStartTickSize sets the size of each 'tick' in hours.
+	ConfigTicksSinceStartTickSize = "TicksSinceStart.TickSize"
+
+	// DefaultTicksSinceStartTickSize is the default number of hours in each 'tick' (24*hour = 1day).
+	DefaultTicksSinceStartTickSize = 24
+)
+
+// Name of this PipelineItem. Uniquely identifies the type, used for mapping keys, etc.
+func (ticks *TicksSinceStart) Name() string {
+	return "TicksSinceStart"
+}
+
+// Provides returns the list of names of entities which are produced by this PipelineItem.
+// Each produced entity will be inserted into `deps` of dependent Consume()-s according
+// to this list. Also used by core.Registry to build the global map of providers.
+func (ticks *TicksSinceStart) Provides() []string {
+	arr := [...]string{DependencyTick}
+	return arr[:]
+}
+
+// Requires returns the list of names of entities which are needed by this PipelineItem.
+// Each requested entity will be inserted into `deps` of Consume(). In turn, those
+// entities are Provides() upstream.
+func (ticks *TicksSinceStart) Requires() []string {
+	return []string{}
+}
+
+// ListConfigurationOptions returns the list of changeable public properties of this PipelineItem.
+func (ticks *TicksSinceStart) ListConfigurationOptions() []core.ConfigurationOption {
+	return []core.ConfigurationOption{{
+		Name:        ConfigTicksSinceStartTickSize,
+		Description: "How long each 'tick' represents in hours.",
+		Flag:        "tick-size",
+		Type:        core.IntConfigurationOption,
+		Default:     DefaultTicksSinceStartTickSize},
+	}
+}
+
+// Configure sets the properties previously published by ListConfigurationOptions().
+func (ticks *TicksSinceStart) Configure(facts map[string]interface{}) error {
+	if val, exists := facts[ConfigTicksSinceStartTickSize].(int); exists {
+		ticks.tickSize = time.Duration(val) * time.Hour
+	} else {
+		// default to 1 day
+		ticks.tickSize = 24 * time.Hour
+	}
+	if ticks.commits == nil {
+		ticks.commits = map[int][]plumbing.Hash{}
+	}
+	facts[FactCommitsByTick] = ticks.commits
```

Comment:
We need to add another fact here which indicates the tick size:
``` go
facts[FactTickSize] = ticks.TickSize
```
Thus `BurndownAnalysis` will be able to read it and save inside `BurndownResult`.

Reply:
also added this in 5b3a96b